scipy/integrate/_cubature.py

```diff
@@ -298,10 +357,45 @@ def cubature(f, a, b, *, rule="gk21", rtol=1e-8, atol=0, max_subdivisions=10000,
 
             rule = ProductNestedFixed([base_rule] * ndim)
```

Comment:
Sorry not directly this PR but it would be great to add test coverage to line 356

Reply:
Great, I'll make a note of this.